addons/project/models/project.py

```diff
@@ -1604,6 +1604,9 @@ def filtered_domain(self, domain):
 
     def copy_data(self, default=None):
         defaults = super().copy_data(default=default)
+        if self.env.user.has_group('project.group_project_user'):
+            BROKEN_FIELDS = ['personal_stage_type_ids']
+            return [{k: v for k, v in default.items() if k not in BROKEN_FIELDS} for default in defaults]
```

Comment:
@Williambraecky
Any idea how to fix the tests without filtering out `personal_stage_type_ids` ?

```py
Build # 13808632 Creating test subuild with config "All at_install" (at_install tests only) 
odoo.sql_db:313
                                    bad query: INSERT INTO project_task_user_rel (task_id, stage_id) VALUES (144, 601) ON CONFLICT DO NOTHING
ERROR: null value in column "user_id" violates not-null constraint
DETAIL:  Failing row contains (115, 144, null, 601, null, null, null, null).

                                  
odoo.addons.project.tests.test_project_flow:40
                                    ERROR: TestProjectFlow.test_project_process_project_manager_duplicate
Traceback (most recent call last):
  File "/data/build/odoo/addons/project/tests/test_project_flow.py", line 40, in test_project_process_project_manager_duplicate
    dogs = pigs.copy()
  File "/data/build/odoo/addons/project/models/project.py", line 473, in copy
    self.map_tasks(project.id)
  File "/data/build/odoo/addons/project/models/project.py", line 454, in map_tasks
    new_task = task.copy(defaults)
  File "/data/build/odoo/addons/project/models/project.py", line 1449, in copy
    return super(Task, self).copy(default)
  File "/data/build/odoo/odoo/models.py", line 4866, in copy
    new = self.with_context(lang=None).create(vals)
  File "<decorator-gen-952>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 412, in _model_create_multi
    return create(self, [arg])
  File "/data/build/odoo/addons/project/models/project.py", line 1682, in create
    tasks = super(Task, self).create(vals_list)
  File "<decorator-gen-168>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/addons/mail/models/mail_thread.py", line 265, in create
    threads = super(MailThread, self).create(vals_list)
  File "<decorator-gen-67>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/odoo/addons/base/models/ir_fields.py", line 613, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 413, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/odoo/models.py", line 4070, in create
    records = self._create(data_list)
  File "/data/build/odoo/odoo/models.py", line 4220, in _create
    field.create([
  File "/data/build/odoo/odoo/fields.py", line 3283, in create
    self.write_batch(record_values, True)
  File "/data/build/odoo/odoo/fields.py", line 3309, in write_batch
    return self.write_real(records_commands_list, create)
  File "/data/build/odoo/odoo/fields.py", line 3879, in write_real
    cr.execute(query, pairs)
  File "<decorator-gen-3>", line 2, in execute
  File "/data/build/odoo/odoo/sql_db.py", line 89, in check
    return f(self, *args, **kwargs)
  File "/data/build/odoo/odoo/sql_db.py", line 310, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.NotNullViolation: null value in column "user_id" violates not-null constraint
DETAIL:  Failing row contains (115, 144, null, 601, null, null, null, null).
```

Reply:
Hello @yelizariev

It seems this will get quite complicated, you probably have noticed the `project_task_user_rel` contains 3 columns.

I think the real solution would be to add a `One2many` on `project.task.stage.personal` computed as sudo (does this work on `x2m`?) and not accessible by anyone, copy that and let the orm compute `personal_stage_type_ids` and make any field depending on that table `copy=False`.
But I don't know how feasable it is or even if we can do it in stable.

On the other hand filtering on that field will not copy the assignees which isn't great either.

Reply:
The filtering is already applied. I just limited to the `personal_stage_type_ids` field. 

Personal stage may need a refactoring, but it's not for stable, I suppose

Reply:
Nevermind, we can put `personal_stage_type_ids` as copy=False, we only care about copying the assignees, having the personal stage being the default one is good enough for the copy, and it is automatically populated during create/write with `_populate_missing_personal_stages`.
In short we don't need to copy that field 😄